citesphere/src/main/java/edu/asu/diging/citesphere/core/service/giles/impl/GilesUploadCheckerImpl.java

```diff
@@ -104,80 +105,42 @@ public void add(ICitation upload) {
-                        processed = mapper.readValue(jsonBody, GilesUpload[].class);
-                    } catch (IOException e) {
-                        logger.error("Could not deserialize response.", e);
-                        upload.setDocumentStatus(GilesStatus.FAILED);
-                        checkedUploads.add(upload);
-                    }
-                    for (GilesUpload processedUpload : processed) {
-                        // giles does not return the progress id again, but we need it
-                        processedUpload.setProgressId(upload.getProgressId());
-                        checkedUploads.add(processedUpload);
-                    }
-                    needsUpdating = true;
-                }
-            }
+            checkUploadStatus(citationKey, null);
+        }
+    }
+    
+    /**
+     * Checks the upload status of all Giles uploads associated with the given citation
+     * and user. For each upload that is still in progress, it queries the Giles service
+     * to retrieve its current status. Depending on the response, uploads may be marked
+     * as COMPLETE, FAILED, or left in progress. If any uploads have updated statuses,
+     * the citation is updated accordingly and, once all uploads are finished, the citation
+     * is removed from the upload queue.
+     *
+     * @param citationKey the unique key identifying the citation whose uploads are being checked
+     * @param user        the user owning the uploads; if null, the uploading user for each
+     *                    upload is looked up via {@link IUserManager#findByUsername(String)}
+     */
+    @Override
+    public void checkUploadStatus(String citationKey, IUser user) {
+        ICitation citation = citationManager.getCitation(citationKey);
+        Set<IGilesUpload> checkedUploads = new HashSet<>();
+        for (IGilesUpload upload : citation.getGilesUploads()) {
+            checkedUploads.addAll(getFileStatus(upload, user));
+        }
 
-            ICitation currentCitation = getCurrentCitation(citation, user);
-            
-            if (needsUpdating) {
-                if (currentCitation != null) {
-                    updateCitation(citation, checkedUploads, user, currentCitation);
-                }
-            }
+        ICitation currentCitation = getCurrentCitation(citation, user);
+        
+        if (currentCitation != null) {
+            updateCitation(citation, checkedUploads, user, currentCitation);
+        }
 
-            int unfinishedUplaods = currentCitation.getGilesUploads().stream()
-                    .filter(u -> !Arrays.asList(GilesStatus.COMPLETE, GilesStatus.FAILED)
-                            .contains(u.getDocumentStatus()))
-                    .collect(Collectors.toList()).size();
-            if (unfinishedUplaods == 0) {
-                uploadQueue.remove(citation.getKey());
-            }
+        int unfinishedUplaods = currentCitation.getGilesUploads().stream()
```

Comment:
so if currentCitation is null in line 134, then this would throw a null pointer, wouldn't it?